packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -211,6 +211,8 @@ export class AmplifyGraphqlApi extends Construct {
       transformParameters: {
         ...defaultTranslationBehavior,
         ...(translationBehavior ?? {}),
+        // migrating from construct -> Gen2 is not supported
+        enableGen2Migration: false,
```

Comment:
1. This is done purposely to prevent `enableGen2Migration` from appearing in the construct docs. I'll add a comment.
2. This parameter will only be used in a Gen 1 project. This flag primarily does two things. 1. add the table mapping to outputs. 2. set all tables removal policy to retain. These steps are necessary to prep the tables to be imported into to a gen 2 project. When the tables are then imported into gen 2 we no longer need this functionality in the gen 2 project so it will always be set to false.

Reply:
Thanks for clarifying. I mis-thought about this as "enable Gen2 _behavior_" rather than "enable Gen2 _migration_". The purpose and use of the flag makes sense to me now.